lib/ancestry/has_ancestry.rb

```diff
@@ -30,6 +30,8 @@ def has_ancestry options = {}
       validates_format_of self.ancestry_column, :with => derive_ancestry_pattern(options[:primary_key_format]), :allow_nil => true
       extend Ancestry::MaterializedPath
 
+      include Ancestry::MaterializedPathPg if ActiveRecord::Base.connection.adapter_name.downcase == "postgresql"
```

Comment:
introduce `update_strategy` that will allow us to opt into this behavior

possibly introduce Ancestry.default_update_strategy defaulted to `:ruby` 

please keep the postgresql check here
but add a check to make sure update strategy is sql

```ruby
class Polygon
  @@sides = 10
  def self.sides
    @@sides
  end
end

puts Polygon.sides # => 10
```

how can we test this? I'm thinking that we should set this to sql in our test suite
so it will become opt out